### PR TITLE
HomeViewController Bugs.

### PR DESCRIPTION
**Pull request by Slavitor**
- Using correct indexPath and returning the array count in the correct delegate method.
- Code refactor.

### CODE REVIEW

File: DVTShowcase/HomeViewController.swift [hunk 1]
```diff
@@ -1,35 +1,28 @@
...
-        self.showcaseAppTableView.delegate = self
-        self.showcaseAppTableView.dataSource = self
-        self.showcaseAppTableView.rowHeight = UITableViewAutomaticDimension
-        self.showcaseAppTableView.estimatedRowHeight = 130
-        self.showcaseAppTableView.register(UINib.init(nibName: "ShowcaseAppTableViewCell", bundle: nil), forCellReuseIdentifier: "CellIdentifier")
+        setUpNavigationBar()
+        setUpTableView()
     }
     
     override func viewDidAppear(_ animated: Bool) {
```
Comment (pjwelcome) [inline DVTShowcase/HomeViewController.swift]: If your not using viewDidApear please remove it

File: DVTShowcase/HomeViewController.swift [hunk 2]
```diff
@@ -39,29 +32,48 @@ class HomeViewController: DVTShowcaseViewController, UITableViewDelegate, UITabl
...
+    func setUpTableView() {
+        self.showcaseAppTableView.delegate = self
+        self.showcaseAppTableView.dataSource = self
+        self.showcaseAppTableView.rowHeight = UITableViewAutomaticDimension
+        self.showcaseAppTableView.estimatedRowHeight = 130
+        self.showcaseAppTableView.register(UINib.init(nibName: String(describing: ShowcaseAppTableViewCell.self), bundle: nil), forCellReuseIdentifier: "CellIdentifier")
+    }
+    
+    private func setUpNavigationBar() {
+        let narBArImageView = UIImageView(frame: CGRect(x: 0, y: 0, width: 30, height: 40))
```
Comment (pjwelcome) [inline DVTShowcase/HomeViewController.swift]: This is not camel case, please fix this

File: DVTShowcase/HomeViewController.swift [hunk 3]
```diff
@@ -39,29 +32,48 @@ class HomeViewController: DVTShowcaseViewController, UITableViewDelegate, UITabl
...
+extension HomeViewController: UITableViewDelegate, UITableViewDataSource {
+    
     func numberOfSections(in tableView: UITableView) -> Int {
-        return showcaseAppArray.count
+        return 1
     }
     
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return 1
+        return  showcaseAppArray.count
```
Comment (pjwelcome) [inline DVTShowcase/HomeViewController.swift]: this naming is bad, please change to something like showcaseApps, we know its an array by its type dont have to mention it in the name of the variable

File: DVTShowcase/HomeViewController.swift [hunk 4]
```diff
@@ -73,37 +85,15 @@ class HomeViewController: DVTShowcaseViewController, UITableViewDelegate, UITabl
         return cell;
     }
     
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        print("indexPathSelected: \(indexPath.row)")
```
Comment (pjwelcome) [inline DVTShowcase/HomeViewController.swift]: please remove this. dont keep prints in your code, only use for testing and remove it.

**Review by pjwelcome — CHANGES_REQUESTED** [review]

**Comment by Slavitor** [comment]
@pjwelcome I didn't write any of this code. All I did was to place the tableView delegate methods into an extension, change "let showcaseApps = showcaseAppArray[indexPath.section]", change return for numberOfRowsInSection from "return 1  " to "showcaseAppArray.count", and vice versa for numberOfSections.

I will make your requested changes.

**Review by pjwelcome — APPROVED** [review]

**Pull request merged**